src/v5_hal/firmware/src/nodes/LiftNode.cpp

```diff
@@ -49,49 +52,125 @@ void LiftNode::setLiftVelocity(int velocity) {
     }
 };
 
-void LiftNode::setLiftPosition(int position, int tolerance) {
+void LiftNode::setLiftPosition(int position, int tolerance=5) {
     m_target_position = position;
     m_tolerance = tolerance;
 };
 
+/**
+ * This should only be called from autonomous 
+ * */
+void LiftNode::setLiftState(LiftState state) {
+    m_lift_state = state;
+}
+
 int LiftNode::getPosition() { // change back to use pot
     return m_left_motor->getPosition();
 }
 
-void LiftNode::updateLiftState() {
+void LiftNode::teleopPeriodic() {
+    m_updateLiftStateTeleop();
+    m_updateLiftPosition();
+        
+        // This is not yet used
+        // case FREE_MOVING: 
+        //     if (m_controller->getController()->get_digital(m_upButton) && 
+        //         !m_controller->getController()->get_digital(m_downButton)) {
+        //         setLiftVoltage(MAX_MOTOR_VOLTAGE); // using voltage here cause we don't know max motor velocity for sure
+        //     } else if (m_controller->getController()->get_digital(m_downButton) && 
+        //         !m_controller->getController()->get_digital(m_upButton)) {
+        //         setLiftVoltage(-MAX_MOTOR_VOLTAGE);
+        //     } else {
+        //         setLiftVelocity(0);
+        //     }   
+        // break;
+};
+
+void LiftNode::autonPeriodic() { 
+    m_updateLiftPosition();
+        
+        // This is not yet used
+        // case FREE_MOVING: 
+        //     // This just allows usage of setLiftPosition() by an autonomous action
+        // break;
+};
+
+void LiftNode::m_updateLiftPosition() {
+    switch (m_lift_state) {
+        case DOWN:
+            setLiftPosition(1000); // UNTESTED
+        break;
+        
+        case UP_FOR_RINGS: 
+            setLiftPosition(900); // UNTESTED
+        break;
+        
+        case FULLY_UP:
+            setLiftPosition(0); // UNTESTED
+        break;
+
+        default:

```

Comment:
If we had a holding state, we could set it inside of here (to prevent the lift from getting into bad states, but that's unlikely)